File: rb/features.rb
# Dingconnect SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DingconnectFeatures
  def self.make_feature(name)
    case name
    when "base"
      DingconnectBaseFeature.new
    when "test"
      DingconnectTestFeature.new
    else
      DingconnectBaseFeature.new
    end
  end
end

File: py/features.py
# Dingconnect SDK feature factory

from feature.base_feature import DingconnectBaseFeature
from feature.test_feature import DingconnectTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DingconnectBaseFeature(),
        "test": lambda: DingconnectTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
